Pipeline: filter: geoip style update (#1680)
fluent/fluent-bit-docs: pipeline/filters/geoip2-filter.md

---
description: Look up Geo data from IP.
---

# GeoIP2 filter

> GeoIP2 Filter lets enrich the incoming data stream using location data from GeoIP2 database.

Reviewer: NB: Missing words

```suggestion
The GeoIP2 filter lets you enrich the incoming data stream with location data from the GeoIP2 database.
```

The `GeoLite2-City.mmdb` database is available from [MaxMind's official site](https://dev.maxmind.com/geoip/geoip2/geolite2/).

> ## Configuration parameters>

Reviewer: Remove extraneous `>` from heading.

```suggestion
## Configuration parameters
```

This plugin supports the following configuration parameters:

| Key | Description |
| :--- | :--- |
| `database` | Path to the GeoIP2 database. |
| `lookup_key` | Field name to process. |
| `record` | Defines the `KEY LOOKUP_KEY VALUE` triplet. |

## Get started

> The following configuration processes incoming `remote_addr`, and appends country information retrieved from GeoLite2 database.

Reviewer: NB: Missing words

```suggestion
The following configuration processes the incoming `remote_addr` and appends country information retrieved from the GeoLite2 database.
```

    Match  *
```

Each `Record` parameter specifies the following triplet:

- `country`: The field name to be added to records.
- `remote_addr`: The lookup key to process.

> - `%{country.names.en}`: The query for GeoIP2 database.

Reviewer: NB: Tighten phrasing

```suggestion
- `%{country.names.en}`: The GeoIP2 database query.
```
